flink-runtime/src/main/java/org/apache/flink/runtime/concurrent/FutureTaskWithException.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.concurrent;
+
+import org.apache.flink.util.function.FunctionUtils;
+import org.apache.flink.util.function.RunnableWithException;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Adapter from {@link FutureTask} to {@link RunnableWithException}.
+ */
+public class FutureTaskWithException<V> implements Future<V>, RunnableWithException {
```

Comment:
Can't we just use FutureTask and wrap the RunnableWithException into a Callable as you did?

Reply:
Unfortunately not. I tried to do just that (actually it was even simpler, since it was enough to pass a `futureTask::run` reference, without any wrapping to convert it into a `RunnableWithException`), but it didn't work since there are `instanceof Future` checks inside the `FutureUtils#cancelRunnableFutures` called by `MailboxProcessor#close`.

Reply:
I didn't get your answer, so let's work with code:

```java
	default @Nonnull Future<Void> submit(@Nonnull RunnableWithException command, String description) {
		FutureTask<Void> future = new FutureTask<>(asCallable(command, null));
		execute(future, description, EMPTY_ARGS);
		return future;
	}
```

Where asCallable is your function from FunctionUtils.

Reply:
```
execute(future, description, EMPTY_ARGS);
```
Compile error
```
Error:(166, 25) java: incompatible types: java.util.concurrent.FutureTask<java.lang.Void> cannot be converted to org.apache.flink.util.function.RunnableWithException
```
Change to:
```
execute(future::run, description, EMPTY_ARGS);
```
results with test failures, since this submitted `RunnableWithException` is no longer a `Future`, so it's not being canceled on `close()`

Reply:
My example did compile for me. Please double-check your setup. Or give me more context. Not sure why it would convert the `FutureTask` to `RunnableWithException`.

Reply:
Btw internally, FutureTask just uses Callables and wraps Runnable into Callable. I'm just proposing to do the same on call-site.

Reply:
resolved after an offline discussion: we decided that `FutureTaskWithException` will stay, but it can be simplified.